### PR TITLE
feat: add beacon polyfill

### PR DESCRIPTION
**Pull request by MoumitaM**
## PR Description

Please include a summary of the change along with the relevant motivation and context.

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: packages/analytics-v1.1/src/utils/constants.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const DEST_SDK_BASE_URL = `${SDK_CDN_BASE_URL}/${CDN_ARCH_VERSION_DIR}/${CDN_INT
 const DEFAULT_DATA_PLANE_EVENTS_BUFFER_TIMEOUT_MS = 10000;
 const INTG_SUFFIX = '_RS';
 const POLYFILL_URL =
-  'https://polyfill.io/v3/polyfill.min.js?features=Number.isNaN%2CURL%2CArray.prototype.find%2CArray.prototype.includes%2CPromise%2CString.prototype.endsWith%2CString.prototype.includes%2CString.prototype.startsWith%2CObject.entries%2CObject.values%2CElement.prototype.dataset%2CString.prototype.replaceAll%2CTextDecoder%2Uint8Array';
+  'https://polyfill.io/v3/polyfill.min.js?features=Number.isNaN%2CURL%2CArray.prototype.find%2CArray.prototype.includes%2CPromise%2CString.prototype.endsWith%2CString.prototype.includes%2CString.prototype.startsWith%2CObject.entries%2CObject.values%2CElement.prototype.dataset%2CString.prototype.replaceAll%2CTextDecoder%2Cnavigator.sendBeacon';
```
Comment (saikumarrs) [inline packages/analytics-v1.1/src/utils/constants.js:23]: `Uint8Array` is not required anymore?

File: packages/analytics-js/__tests__/components/capabilitiesManager/detection/dom.test.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ describe('Capabilities Detection - DOM', () => {
     expect(isDatasetAvailable()).toBeTruthy();
   });
   it('should detect not supported javascript features', () => {
+    navigator.sendBeacon = jest.fn();
```
Comment (bardisg) [inline packages/analytics-js/__tests__/components/capabilitiesManager/detection/dom.test.ts:11]: you can configure this one globaly too if you like for all unit tests

File: packages/analytics-v1.1/src/utils/constants.js (hunk 1 above)
Comment (MoumitaM) [reply packages/analytics-v1.1/src/utils/constants.js:23]: It is not supported anymore

File: packages/analytics-js/__tests__/components/capabilitiesManager/detection/dom.test.ts (hunk 2 above)
Comment (MoumitaM) [reply packages/analytics-js/__tests__/components/capabilitiesManager/detection/dom.test.ts]: Done

File: packages/analytics-v1.1/src/utils/constants.js (hunk 1 above)
Comment (saikumarrs) [reply packages/analytics-v1.1/src/utils/constants.js:23]: @MoumitaM, we need to check whether these polyfill query strings are valid individually. For example, `TextDecoder` is not supported.

File: packages/analytics-js/src/components/capabilitiesManager/detection/dom.ts [hunk 3]
```diff
@@ -30,6 +30,8 @@ const legacyJSEngineRequiredPolyfills: Record<string, () => boolean> = {
   requestAnimationFrame: () => !isFunction(globalThis.requestAnimationFrame),
   cancelAnimationFrame: () => !isFunction(globalThis.cancelAnimationFrame),
```
Comment (saikumarrs) [inline packages/analytics-js/src/components/capabilitiesManager/detection/dom.ts:31]: Similar to the other comment, for example `requestAnimationFrame`, `URLSearchParams` are not supported anymore.
We need to check the supported polyfill groups like `default`, `es6`, `dom`, etc if they can fill these gaps for us.

CC @bardisg

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: packages/analytics-js/src/components/capabilitiesManager/detection/dom.ts (hunk 3 above)
Comment (saikumarrs) [reply packages/analytics-js/src/components/capabilitiesManager/detection/dom.ts:31]: Moreover, with polyfill groups, the URL also becomes more straightforward for us.

Comment (saikumarrs) [reply packages/analytics-js/src/components/capabilitiesManager/detection/dom.ts:31]: Let's use `https://caniuse.com/` for checking the support. https://developer.mozilla.org/en-US/ seems a bit unreliable for some features.

Comment (MoumitaM) [reply packages/analytics-js/src/components/capabilitiesManager/detection/dom.ts:31]: Will create a different ticket for that. Let's not block this one.

**Review by saikumarrs — APPROVED** [review]
Unblocking this PR as @MoumitaM will create a new task with all the requested changes and investigations.

**Pull request merged**